src/bluefinClient.ts

```diff
@@ -1538,6 +1540,298 @@ export class BluefinClient {
     return this.contractCalls.withdrawAllFromMarginBankContractCall();
   };
 
+  closeAllDelistedPositionsAndWithdrawMargin = async (args?: {
+    gasBudget?: number;
+    sponsor?: boolean;
+    dryRunOnly?: boolean;
+  }) => {
+    try {
+      // get exchange info of all markets to read the status
+      const exchangeInfo = await this.getExchangeInfo();
+      const delistedMarkets = filterDelistedMarkets(exchangeInfo);
+
+      if (delistedMarkets.length <= 0) {
+        return {
+          ok: false,
+          code: 400,
+          data: "",
+          message: "No delisted markets",
+        };
+      }
+
+      //get user positions
+      const userPositions = await this.getUserPosition({});
+
+      let delistedUserPositionsSymbols: string[] = [];
+      //get delisted market positions
+      if (userPositions.data && userPositions.data.length > 0) {
+        delistedUserPositionsSymbols = userPositions.data
+          .filter((position) => {
+            return delistedMarkets.includes(position.symbol);
+          })
+          .map((position) => position.symbol);
+      }
+
+      if (delistedUserPositionsSymbols.length <= 0) {
+        return {
+          ok: false,
+          code: 400,
+          data: "",
+          message: "No delisted positions",
+        };
+      }
+
+      // cancel open orders for delisted markets
+      await Promise.all(
+        delistedUserPositionsSymbols.map((symbol) =>
+          this.cancelAllOpenOrders(symbol)
+        )
+      );
+
+      // create PTB and execute
+      if (args?.sponsor && args?.sponsor == true) {
+        const sponsoredPayload =
+          await this.contractCalls.closeAllPositionsAndWithdrawMarginPTB(
+            delistedUserPositionsSymbols,
+            args
+          );
+
+        const sponsorTxResponse = await this.signAndExecuteSponsoredTx(
+          sponsoredPayload
+        );
+
+        if (sponsorTxResponse?.ok) {
+          return {
+            ok: true,
+            code: 200,
+            data: sponsorTxResponse,
+            message: "Closed delisted Positions and Withdraw Margin Successful",
+          };
+        }
+
+        if (
+          !sponsorTxResponse?.ok &&
+          sponsorTxResponse?.message !== USER_REJECTED_MESSAGE
+        ) {
+          return this.closeAllDelistedPositionsWithdrawSwapAndDepositToPro({
```

Comment:
Shouldn't this be `closeAllPositionsAndWithdrawMarginPTB`? Or just `closeAllDelistedPositionsAndWithdrawMargin`